### PR TITLE
Fix #257: Implement limit for max attempts of document upload

### LINKED ISSUE #257: Implement limit for max attempts of document upload
When the max. attempt count for document upload is exceeded, the identity verification should fail. The default max. attempt count is 5.

### PR DESCRIPTION
**Pull request by romanstrobl**
This PR builds on changes introduced in: https://github.com/wultra/enrollment-server/pull/266 and adds a limit for maximum number of attempts for document upload.

### CODE REVIEW

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationLimitService.java [hunk 1]
```diff
@@ -107,4 +119,25 @@ public void checkIdentityVerificationLimit(OwnerId ownerId) throws PowerAuthClie
...
+    /**
+     * Check the limit for maximum number of document uploads.
+     * @param ownerId Owner identifier.
+     * @throws IdentityVerificationLimitException Thrown in case document upload limit is reached.
+     * @throws RemoteCommunicationException Thrown when communication with PowerAuth server fails.
+     * @throws IdentityVerificationException Thrown in case identity verification is invalid.
+     * @throws OnboardingProcessLimitException Thrown when maximum failed attempts for identity verification have been reached.
+     */
+    public void checkDocumentUploadLimit(OwnerId ownerId, IdentityVerificationEntity identityVerification) throws IdentityVerificationLimitException, RemoteCommunicationException, IdentityVerificationException, OnboardingProcessLimitException {
+        List<DocumentVerificationEntity> documentVerificationsFailed = documentVerificationRepository.findAllDocumentVerifications(identityVerification, DocumentStatus.ALL_UNSUCCESSFUL);
```
Comment (saalistaja) [inline enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationLimitService.java:132]: I see a little contrast between naming `failed` and `ALL_UNSUCCESSFUL`. Maybe we the constant `DocumentStatus.ALL_UNSUCCESSFUL` could be renamed to `ALL_FAILED` to keep consistent.

**Review by saalistaja — APPROVED** [review]
Nice work. This should work as expected!

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationLimitService.java (hunk 1 above)
Comment (romanstrobl) [reply enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationLimitService.java:132]: Yes, we can change that.

File: enrollment-server-domain-model/src/main/java/com/wultra/app/enrollmentserver/model/enumeration/DocumentStatus.java [hunk 2]
```diff
@@ -65,12 +65,21 @@ public enum DocumentStatus {
...
     public static final List<DocumentStatus> ALL_NOT_FINISHED = ImmutableList.of(
             UPLOAD_IN_PROGRESS,
             VERIFICATION_PENDING,
             VERIFICATION_IN_PROGRESS
     );
 
+    /**
+     * All unsuccessful statuses.
+     */
+    public static final List<DocumentStatus> ALL_FAILED = ImmutableList.of(
```
Comment (banterCZ) [inline enrollment-server-domain-model/src/main/java/com/wultra/app/enrollmentserver/model/enumeration/DocumentStatus.java:79]: Have you considered to introduce a property to the enum and filter values based on it? I mean, it is more error proof when we add another enum values not to forget edit this list.

Comment (romanstrobl) [reply enrollment-server-domain-model/src/main/java/com/wultra/app/enrollmentserver/model/enumeration/DocumentStatus.java:79]: We can improve it in a separate issue, if you feel strongly about it. I cannot think of any other states right now.

Comment (banterCZ) [reply enrollment-server-domain-model/src/main/java/com/wultra/app/enrollmentserver/model/enumeration/DocumentStatus.java:79]: I can live with that so far ;) I just wanted to mention a way, how it is possible to work with enums next time.

**Pull request merged**